frame/identity/src/lib.rs

```diff
@@ -298,28 +308,85 @@ pub struct IdentityInfo {
 	pub twitter: Data,
 }
 
+#[cfg(not(feature = "std"))]
+impl<FieldLimit: Get<u32>> Debug for IdentityInfo<FieldLimit> {
+	fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+		write!(f, "<wasm:stripped>")
+	}
+}
+
+#[cfg(feature = "std")]
+impl<FieldLimit: Get<u32>> Debug for IdentityInfo<FieldLimit> {
+	fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+		f.debug_struct("IdentityInfo")
+			.field("additional", &self.additional)
+			.field("display", &self.display)
+			.field("legal", &self.legal)
+			.field("web", &self.web)
+			.field("riot", &self.riot)
+			.field("email", &self.email)
+			.field("pgp_fingerprint", &self.pgp_fingerprint)
+			.field("image", &self.image)
+			.field("twitter", &self.twitter)
+			.finish()
+	}
+}
+
+impl<FieldLimit: Get<u32>> PartialEq for IdentityInfo<FieldLimit> {
+	fn eq(&self, other: &IdentityInfo<FieldLimit>) -> bool {
+		self.additional == other.additional
+			&& self.display == other.display
+			&& self.legal == other.legal
+			&& self.web == other.web
+			&& self.riot == other.riot
+			&& self.email == other.email
+			&& self.image == other.image
+			&& self.twitter == other.twitter
```

Comment:
why leave out the `pgp_fingerprint`?

Reply:
(and relatedly: why not derive?)

Reply:
For some strange reason, derive was not working properly once I added a type parameter to `IdentityInfo` -- it requires `FieldLimit` to also be `Clone`, `Debug` and `PartialEq`, and I can't figure out why it was doing so just with this specific struct and not others.

Reply:
Yeah, IIRC this was the reason for introducing the `*NoBound` family of derives. There is `RuntimeDebugNoBound` which should do what you want (strip in `no_std`, derive Debug in std and ignore the generic type).